R/createAttributePlan.R

```diff
@@ -0,0 +1,190 @@
+  
+  df_plan <- data.frame(PD = pd, PA = plan@paccept)
+  df_plan <- na.omit(df_plan)
+  if (nrow(df_plan) == 0) {
+    jaspContainer$setError(sprintf("No valid values found in the plan. Check the inputs."))
+    return ()
+  }
+  # Fill the output tables for the created plan.
+  .attributePlanTable(jaspContainer, depend_vars, aql, rql, df_plan$PA[df_plan$PD == aql], df_plan$PA[df_plan$PD == rql], n, c, r)
+
+  # Plan summary
+  if (options$showSummary) {
+    getSummary(jaspContainer, pos=4, c(depend_vars, "showSummary"), df_plan)
+  }
+
+  # OC Curve
+  if (options$showOCCurve) {
+    getOCCurve(jaspContainer, pos=5, c(depend_vars, "showOCCurve"), df_plan)
+  }
+
+  # AOQ Curve (for plans with rectification)
+  if (options$showAOQCurve) {
+    getAOQCurve(jaspContainer, pos=6, "showAOQCurve", df_plan, options, "", n)
+    if (jaspContainer$getError()) {
+      return ()
+    }
+  }
+
+  # ATI Curve (for plans with rectification)
+  if (options$showATICurve) {
+    getATICurve(jaspContainer, pos=7, "showATICurve", df_plan, options, "", n)
+    if (jaspContainer$getError()) {
+      return ()
+    }
+  }
+}
+
+##----------------------------------------------------------------
+##             Create and fill the output table(s).             --
+##----------------------------------------------------------------
+#' @param jaspContainer {list} A functional grouping of different output elements such as plots, tables, etc.
+#' @param depend_vars {vector} Names of variables on which the output element depends.
+#' @param aql {numeric} Acceptable Quality Level (AQL), specified as the proportion (0 to 1) of non-conforming items.
+#' @param rql {numeric} Rejectable Quality Level (RQL), specified as the proportion (0 to 1) of non-conforming items.
+#' @param pa_prod {numeric} Minimum probability (0 to 1) of accepting the lot at Acceptable Quality Level.
+#' @param pa_cons {numeric} Maximum probability (0 to 1) of accepting the lot at Rejectable Quality Level.
+#' @param n {numeric} Sample size for the plan.
+#' @param c {numeric} Acceptance number for the plan.
+#' @param r {numeric} Rejection number for the plan.
+##----------------------------------------------------------------
+.attributePlanTable <- function(jaspContainer, depend_vars, aql, rql, pa_prod, pa_cons, n, c, r) {
+  # Fill the table with sample size and acceptance number
+  plan_table <- jaspContainer[["findPlanTable"]]
+  plan_table[["col_2"]] <- c(n, c)
+
+  # Fill the table with acceptance and rejection probabilities for AQL and RQL
+  prob_table <- jaspContainer[["findProbTable"]]
+  prob_table[["col_2"]] <- c(aql, rql)
+  prob_table[["col_3"]] <- c(pa_prod, pa_cons)
+  prob_table[["col_4"]] <- c(1-pa_prod, 1-pa_cons)
+
+  # Description of the sampling plan:
+  if (is.null(jaspContainer[["description"]])) {
+    description <- createJaspHtml(text = sprintf("If the number of defective items out of %d sampled is <= %d, accept the lot. Reject otherwise.", n, c), position = 3)
```

Comment:
```suggestion
    description <- createJaspHtml(text = gettextf("If the number of defective items out of %1$d sampled is <= %2$d, accept the lot. Reject otherwise.", n, c), position = 3)
```
The `1$` and `2$` are positional placeholders so that a translator could change the order of the interpolated variables if necessary. This is only necessary to do when there are 2 or more interpolated variables. You can also use it to interpolate the same value multiple times.